pkg/controller/jaeger/jaeger_controller.go

```diff
@@ -237,50 +256,54 @@ func (r *ReconcileJaeger) apply(jaeger v1.Jaeger, str strategy.S) (v1.Jaeger, er
 	}
 
 	// storage dependencies have to be deployed after ES is ready
-	if err := r.handleDependencies(str); err != nil {
-		return jaeger, errors.Wrap(err, "failed to handler dependencies")
+	if err := r.handleDependencies(ctx, str); err != nil {
+		return handleErrorWithJaeger(jaeger, err, span)
 	}
 
-	if err := r.applyAccounts(jaeger, str.Accounts()); err != nil {
-		return jaeger, err
+	if err := r.applyAccounts(ctx, jaeger, str.Accounts()); err != nil {
+		return handleErrorWithJaeger(jaeger, err, span)
 	}
 
-	if err := r.applyClusterRoleBindingBindings(jaeger, str.ClusterRoleBindings()); err != nil {
-		return jaeger, err
+	if err := r.applyClusterRoleBindingBindings(ctx, jaeger, str.ClusterRoleBindings()); err != nil {
+		return handleErrorWithJaeger(jaeger, err, span)
 	}
 
-	if err := r.applyConfigMaps(jaeger, str.ConfigMaps()); err != nil {
-		return jaeger, err
+	if err := r.applyConfigMaps(ctx, jaeger, str.ConfigMaps()); err != nil {
+		return handleErrorWithJaeger(jaeger, err, span)
 	}
 
-	if err := r.applyCronJobs(jaeger, str.CronJobs()); err != nil {
-		return jaeger, err
+	if err := r.applyCronJobs(ctx, jaeger, str.CronJobs()); err != nil {
+		return handleErrorWithJaeger(jaeger, err, span)
 	}
 
-	if err := r.applyDaemonSets(jaeger, str.DaemonSets()); err != nil {
-		return jaeger, err
+	if err := r.applyDaemonSets(ctx, jaeger, str.DaemonSets()); err != nil {
+		return handleErrorWithJaeger(jaeger, err, span)
 	}
 
 	// seems counter intuitive to have services created *before* deployments,
 	// but some resources used by deployments are created by services, such as TLS certs
 	// for the oauth proxy, if one is used
-	if err := r.applyServices(jaeger, str.Services()); err != nil {
-		return jaeger, err
+	if err := r.applyServices(ctx, jaeger, str.Services()); err != nil {
+		return handleErrorWithJaeger(jaeger, err, span)
 	}
 
-	if err := r.applyDeployments(jaeger, str.Deployments()); err != nil {
-		return jaeger, err
+	if err := r.applyDeployments(ctx, jaeger, str.Deployments()); err != nil {
+		return handleErrorWithJaeger(jaeger, err, span)
 	}
 
 	if strings.EqualFold(viper.GetString("platform"), v1.FlagPlatformOpenShift) {
-		if err := r.applyRoutes(jaeger, str.Routes()); err != nil {
-			return jaeger, err
+		if err := r.applyRoutes(ctx, jaeger, str.Routes()); err != nil {
+			return handleErrorWithJaeger(jaeger, err, span)
 		}
 	} else {
-		if err := r.applyIngresses(jaeger, str.Ingresses()); err != nil {
-			return jaeger, err
+		if err := r.applyIngresses(ctx, jaeger, str.Ingresses()); err != nil {
+			return handleErrorWithJaeger(jaeger, err, span)
 		}
 	}
 
 	return jaeger, nil
 }
+
+func handleErrorWithJaeger(jaeger v1.Jaeger, err error, span apitrace.Span) (v1.Jaeger, error) {
+	return jaeger, tracing.HandleError(err, span)
```

Comment:
Is this function necessary? As everywhere else in the code just replaces `err` with `tracing.HandleError(err, span)`?